{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "c331c848",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><style>\n",
       ".dataframe > thead > tr,\n",
       ".dataframe > tbody > tr {\n",
       "  text-align: right;\n",
       "  white-space: pre-wrap;\n",
       "}\n",
       "</style>\n",
       "<small>shape: (5, 10)</small><table border=\"1\" class=\"dataframe\"><thead><tr><th>user_id</th><th>item_id</th><th>action_type_view_description</th><th>action_type_to_cart</th><th>action_type_page_view</th><th>action_type_favorite</th><th>action_type_unfavorite</th><th>action_type_review_view</th><th>action_type_remove</th><th>last_status</th></tr><tr><td>i64</td><td>i64</td><td>i32</td><td>i32</td><td>i32</td><td>i32</td><td>i32</td><td>i32</td><td>i32</td><td>i8</td></tr></thead><tbody><tr><td>4721531</td><td>103451030</td><td>0</td><td>0</td><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr><tr><td>4233180</td><td>141653295</td><td>0</td><td>0</td><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr><tr><td>4602951</td><td>193164064</td><td>0</td><td>1</td><td>0</td><td>1</td><td>1</td><td>0</td><td>1</td><td>0</td></tr><tr><td>1729611</td><td>105254873</td><td>0</td><td>0</td><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr><tr><td>507430</td><td>80041107</td><td>0</td><td>0</td><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr></tbody></table></div>"
      ],
      "text/plain": [
       "shape: (5, 10)\n",
       "┌─────────┬───────────┬────────────┬───────────┬───┬───────────┬───────────┬───────────┬───────────┐\n",
       "│ user_id ┆ item_id   ┆ action_typ ┆ action_ty ┆ … ┆ action_ty ┆ action_ty ┆ action_ty ┆ last_stat │\n",
       "│ ---     ┆ ---       ┆ e_view_des ┆ pe_to_car ┆   ┆ pe_unfavo ┆ pe_review ┆ pe_remove ┆ us        │\n",
       "│ i64     ┆ i64       ┆ cription   ┆ t         ┆   ┆ rite      ┆ _view     ┆ ---       ┆ ---       │\n",
       "│         ┆           ┆ ---        ┆ ---       ┆   ┆ ---       ┆ ---       ┆ i32       ┆ i8        │\n",
       "│         ┆           ┆ i32        ┆ i32       ┆   ┆ i32       ┆ i32       ┆           ┆           │\n",
       "╞═════════╪═══════════╪════════════╪═══════════╪═══╪═══════════╪═══════════╪═══════════╪═══════════╡\n",
       "│ 4721531 ┆ 103451030 ┆ 0          ┆ 0         ┆ … ┆ 0         ┆ 0         ┆ 0         ┆ 0         │\n",
       "│ 4233180 ┆ 141653295 ┆ 0          ┆ 0         ┆ … ┆ 0         ┆ 0         ┆ 0         ┆ 0         │\n",
       "│ 4602951 ┆ 193164064 ┆ 0          ┆ 1         ┆ … ┆ 1         ┆ 0         ┆ 1         ┆ 0         │\n",
       "│ 1729611 ┆ 105254873 ┆ 0          ┆ 0         ┆ … ┆ 0         ┆ 0         ┆ 0         ┆ 0         │\n",
       "│ 507430  ┆ 80041107  ┆ 0          ┆ 0         ┆ … ┆ 0         ┆ 0         ┆ 0         ┆ 0         │\n",
       "└─────────┴───────────┴────────────┴───────────┴───┴───────────┴───────────┴───────────┴───────────┘"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import polars as pl\n",
    "df = pl.read_parquet(\"data/grouped_batch_1.parquet\")\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "c37b9283",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.preprocessing import StandardScaler\n",
    "import numpy as np\n",
    "\n",
    "feature_cols = [\n",
    "    \"action_type_view_description\",\n",
    "    \"action_type_to_cart\",\n",
    "    \"action_type_page_view\",\n",
    "    \"action_type_favorite\",\n",
    "    \"action_type_unfavorite\",\n",
    "    \"action_type_review_view\",\n",
    "    \"action_type_remove\"\n",
    "]\n",
    "\n",
    "X = df.select(feature_cols).to_numpy()\n",
    "y = df[\"last_status\"].to_numpy()\n",
    "\n",
    "scaler = StandardScaler()\n",
    "X_scaled = scaler.fit_transform(X)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "8af159f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.model_selection import train_test_split\n",
    "X_train, X_test, y_train, y_test = train_test_split(\n",
    "    X_scaled, y, test_size=0.2, random_state=42\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "ea7c5d65",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mKeyboardInterrupt\u001b[39m                         Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[23]\u001b[39m\u001b[32m, line 5\u001b[39m\n\u001b[32m      2\u001b[39m \u001b[38;5;28;01mfrom\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[34;01msklearn\u001b[39;00m\u001b[34;01m.\u001b[39;00m\u001b[34;01mmetrics\u001b[39;00m\u001b[38;5;250m \u001b[39m\u001b[38;5;28;01mimport\u001b[39;00m classification_report\n\u001b[32m      4\u001b[39m model_gb = GradientBoostingClassifier()\n\u001b[32m----> \u001b[39m\u001b[32m5\u001b[39m \u001b[43mmodel_gb\u001b[49m\u001b[43m.\u001b[49m\u001b[43mfit\u001b[49m\u001b[43m(\u001b[49m\u001b[43mX_train\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43my_train\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m      7\u001b[39m y_pred_gb = model_gb.predict(X_test)\n\u001b[32m      9\u001b[39m \u001b[38;5;28mprint\u001b[39m(classification_report(y_test, y_pred_gb))\n",
      "\u001b[36mFile \u001b[39m\u001b[32m/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/sklearn/base.py:1365\u001b[39m, in \u001b[36m_fit_context.<locals>.decorator.<locals>.wrapper\u001b[39m\u001b[34m(estimator, *args, **kwargs)\u001b[39m\n\u001b[32m   1358\u001b[39m     estimator._validate_params()\n\u001b[32m   1360\u001b[39m \u001b[38;5;28;01mwith\u001b[39;00m config_context(\n\u001b[32m   1361\u001b[39m     skip_parameter_validation=(\n\u001b[32m   1362\u001b[39m         prefer_skip_nested_validation \u001b[38;5;129;01mor\u001b[39;00m global_skip_validation\n\u001b[32m   1363\u001b[39m     )\n\u001b[32m   1364\u001b[39m ):\n\u001b[32m-> \u001b[39m\u001b[32m1365\u001b[39m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mfit_method\u001b[49m\u001b[43m(\u001b[49m\u001b[43mestimator\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m*\u001b[49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/sklearn/ensemble/_gb.py:787\u001b[39m, in \u001b[36mBaseGradientBoosting.fit\u001b[39m\u001b[34m(self, X, y, sample_weight, monitor)\u001b[39m\n\u001b[32m    784\u001b[39m     \u001b[38;5;28mself\u001b[39m._resize_state()\n\u001b[32m    786\u001b[39m \u001b[38;5;66;03m# fit the boosting stages\u001b[39;00m\n\u001b[32m--> \u001b[39m\u001b[32m787\u001b[39m n_stages = \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_fit_stages\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    788\u001b[39m \u001b[43m    \u001b[49m\u001b[43mX_train\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    789\u001b[39m \u001b[43m    \u001b[49m\u001b[43my_train\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    790\u001b[39m \u001b[43m    \u001b[49m\u001b[43mraw_predictions\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    791\u001b[39m \u001b[43m    \u001b[49m\u001b[43msample_weight_train\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    792\u001b[39m \u001b[43m    \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_rng\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    793\u001b[39m \u001b[43m    \u001b[49m\u001b[43mX_val\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    794\u001b[39m \u001b[43m    \u001b[49m\u001b[43my_val\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    795\u001b[39m \u001b[43m    \u001b[49m\u001b[43msample_weight_val\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    796\u001b[39m \u001b[43m    \u001b[49m\u001b[43mbegin_at_stage\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    797\u001b[39m \u001b[43m    \u001b[49m\u001b[43mmonitor\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    798\u001b[39m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    800\u001b[39m \u001b[38;5;66;03m# change shape of arrays after fit (early-stopping or additional ests)\u001b[39;00m\n\u001b[32m    801\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m n_stages != \u001b[38;5;28mself\u001b[39m.estimators_.shape[\u001b[32m0\u001b[39m]:\n",
      "\u001b[36mFile \u001b[39m\u001b[32m/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/sklearn/ensemble/_gb.py:883\u001b[39m, in \u001b[36mBaseGradientBoosting._fit_stages\u001b[39m\u001b[34m(self, X, y, raw_predictions, sample_weight, random_state, X_val, y_val, sample_weight_val, begin_at_stage, monitor)\u001b[39m\n\u001b[32m    876\u001b[39m         initial_loss = factor * \u001b[38;5;28mself\u001b[39m._loss(\n\u001b[32m    877\u001b[39m             y_true=y_oob_masked,\n\u001b[32m    878\u001b[39m             raw_prediction=raw_predictions[~sample_mask],\n\u001b[32m    879\u001b[39m             sample_weight=sample_weight_oob_masked,\n\u001b[32m    880\u001b[39m         )\n\u001b[32m    882\u001b[39m \u001b[38;5;66;03m# fit next stage of trees\u001b[39;00m\n\u001b[32m--> \u001b[39m\u001b[32m883\u001b[39m raw_predictions = \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_fit_stage\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    884\u001b[39m \u001b[43m    \u001b[49m\u001b[43mi\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    885\u001b[39m \u001b[43m    \u001b[49m\u001b[43mX\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    886\u001b[39m \u001b[43m    \u001b[49m\u001b[43my\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    887\u001b[39m \u001b[43m    \u001b[49m\u001b[43mraw_predictions\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    888\u001b[39m \u001b[43m    \u001b[49m\u001b[43msample_weight\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    889\u001b[39m \u001b[43m    \u001b[49m\u001b[43msample_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    890\u001b[39m \u001b[43m    \u001b[49m\u001b[43mrandom_state\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    891\u001b[39m \u001b[43m    \u001b[49m\u001b[43mX_csc\u001b[49m\u001b[43m=\u001b[49m\u001b[43mX_csc\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    892\u001b[39m \u001b[43m    \u001b[49m\u001b[43mX_csr\u001b[49m\u001b[43m=\u001b[49m\u001b[43mX_csr\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    893\u001b[39m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    895\u001b[39m \u001b[38;5;66;03m# track loss\u001b[39;00m\n\u001b[32m    896\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m do_oob:\n",
      "\u001b[36mFile \u001b[39m\u001b[32m/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/sklearn/ensemble/_gb.py:495\u001b[39m, in \u001b[36mBaseGradientBoosting._fit_stage\u001b[39m\u001b[34m(self, i, X, y, raw_predictions, sample_weight, sample_mask, random_state, X_csc, X_csr)\u001b[39m\n\u001b[32m    493\u001b[39m \u001b[38;5;66;03m# update tree leaves\u001b[39;00m\n\u001b[32m    494\u001b[39m X_for_tree_update = X_csr \u001b[38;5;28;01mif\u001b[39;00m X_csr \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;28;01melse\u001b[39;00m X\n\u001b[32m--> \u001b[39m\u001b[32m495\u001b[39m \u001b[43m_update_terminal_regions\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    496\u001b[39m \u001b[43m    \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_loss\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    497\u001b[39m \u001b[43m    \u001b[49m\u001b[43mtree\u001b[49m\u001b[43m.\u001b[49m\u001b[43mtree_\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    498\u001b[39m \u001b[43m    \u001b[49m\u001b[43mX_for_tree_update\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    499\u001b[39m \u001b[43m    \u001b[49m\u001b[43my\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    500\u001b[39m \u001b[43m    \u001b[49m\u001b[43mneg_g_view\u001b[49m\u001b[43m[\u001b[49m\u001b[43m:\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mk\u001b[49m\u001b[43m]\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    501\u001b[39m \u001b[43m    \u001b[49m\u001b[43mraw_predictions\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    502\u001b[39m \u001b[43m    \u001b[49m\u001b[43msample_weight\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    503\u001b[39m \u001b[43m    \u001b[49m\u001b[43msample_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    504\u001b[39m \u001b[43m    \u001b[49m\u001b[43mlearning_rate\u001b[49m\u001b[43m=\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43mlearning_rate\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    505\u001b[39m \u001b[43m    \u001b[49m\u001b[43mk\u001b[49m\u001b[43m=\u001b[49m\u001b[43mk\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    506\u001b[39m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    508\u001b[39m \u001b[38;5;66;03m# add tree to ensemble\u001b[39;00m\n\u001b[32m    509\u001b[39m \u001b[38;5;28mself\u001b[39m.estimators_[i, k] = tree\n",
      "\u001b[36mFile \u001b[39m\u001b[32m/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/sklearn/ensemble/_gb.py:253\u001b[39m, in \u001b[36m_update_terminal_regions\u001b[39m\u001b[34m(loss, tree, X, y, neg_gradient, raw_prediction, sample_weight, sample_mask, learning_rate, k)\u001b[39m\n\u001b[32m    249\u001b[39m \u001b[38;5;28;01mfor\u001b[39;00m leaf \u001b[38;5;129;01min\u001b[39;00m np.nonzero(tree.children_left == TREE_LEAF)[\u001b[32m0\u001b[39m]:\n\u001b[32m    250\u001b[39m     indices = np.nonzero(masked_terminal_regions == leaf)[\n\u001b[32m    251\u001b[39m         \u001b[32m0\u001b[39m\n\u001b[32m    252\u001b[39m     ]  \u001b[38;5;66;03m# of terminal regions\u001b[39;00m\n\u001b[32m--> \u001b[39m\u001b[32m253\u001b[39m     y_ = \u001b[43my\u001b[49m\u001b[43m.\u001b[49m\u001b[43mtake\u001b[49m\u001b[43m(\u001b[49m\u001b[43mindices\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43maxis\u001b[49m\u001b[43m=\u001b[49m\u001b[32;43m0\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[32m    254\u001b[39m     sw = \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;28;01mif\u001b[39;00m sample_weight \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;28;01melse\u001b[39;00m sample_weight[indices]\n\u001b[32m    255\u001b[39m     update = compute_update(y_, indices, neg_gradient, raw_prediction, k)\n",
      "\u001b[31mKeyboardInterrupt\u001b[39m: "
     ]
    }
   ],
   "source": [
    "from sklearn.ensemble import GradientBoostingClassifier\n",
    "from sklearn.metrics import classification_report\n",
    "\n",
    "model_gb = GradientBoostingClassifier()\n",
    "model_gb.fit(X_train, y_train)\n",
    "\n",
    "y_pred_gb = model_gb.predict(X_test)\n",
    "\n",
    "print(classification_report(y_test, y_pred_gb))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "60a624fa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0       1.00      0.98      0.99  14827301\n",
      "           1       0.35      0.65      0.45    188431\n",
      "\n",
      "    accuracy                           0.98  15015732\n",
      "   macro avg       0.67      0.82      0.72  15015732\n",
      "weighted avg       0.99      0.98      0.98  15015732\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.metrics import classification_report\n",
    "\n",
    "\n",
    "model_log_reg = LogisticRegression(\n",
    "    class_weight={0: 1, 1: 40},     # учитываем дисбаланс\n",
    "    max_iter=1000,\n",
    "    C=0.5,                       # сила регуляризации (меньше — сильнее)\n",
    "    penalty=\"l2\",\n",
    "    solver=\"lbfgs\"\n",
    ")\n",
    "model_log_reg.fit(X_train, y_train)\n",
    "\n",
    "\n",
    "y_proba = model_log_reg.predict_proba(X_test)[:, 1]\n",
    "\n",
    "\n",
    "threshold = 0.9\n",
    "y_pred_custom = (y_proba >= threshold).astype(int)\n",
    "\n",
    "\n",
    "print(classification_report(y_test, y_pred_custom))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "5f316119",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Лучший порог: 0.90, F1: 0.4518\n"
     ]
    }
   ],
   "source": [
    "y_pred_proba = model_log_reg.predict_proba(X_test)[:, 1]\n",
    "\n",
    "from sklearn.metrics import f1_score, precision_score, recall_score\n",
    "\n",
    "best_f1, best_t = 0, 0\n",
    "for t in [0.1 * i for i in range(1, 10)]:\n",
    "    y_pred = (y_pred_proba > t).astype(int)\n",
    "    f1 = f1_score(y_test, y_pred)\n",
    "    if f1 > best_f1:\n",
    "        best_f1, best_t = f1, t\n",
    "\n",
    "print(f\"Лучший порог: {best_t:.2f}, F1: {best_f1:.4f}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3f16a99e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0:\tlearn: 0.7000081\ttest: 0.6998447\tbest: 0.6998447 (0)\ttotal: 483ms\tremaining: 8m 2s\n",
      "100:\tlearn: 0.6993736\ttest: 0.6992311\tbest: 0.6999405 (1)\ttotal: 33.9s\tremaining: 5m 1s\n",
      "200:\tlearn: 0.6993764\ttest: 0.6991920\tbest: 0.6999405 (1)\ttotal: 1m 7s\tremaining: 4m 29s\n",
      "300:\tlearn: 0.6996745\ttest: 0.6994762\tbest: 0.6999405 (1)\ttotal: 1m 41s\tremaining: 3m 54s\n",
      "400:\tlearn: 0.6996739\ttest: 0.6994615\tbest: 0.6999405 (1)\ttotal: 2m 14s\tremaining: 3m 20s\n",
      "500:\tlearn: 0.6995886\ttest: 0.6993561\tbest: 0.6999405 (1)\ttotal: 2m 47s\tremaining: 2m 47s\n",
      "600:\tlearn: 0.6994948\ttest: 0.6992271\tbest: 0.6999405 (1)\ttotal: 3m 25s\tremaining: 2m 16s\n",
      "700:\tlearn: 0.6995271\ttest: 0.6992497\tbest: 0.6999405 (1)\ttotal: 3m 57s\tremaining: 1m 41s\n",
      "800:\tlearn: 0.6995538\ttest: 0.6992742\tbest: 0.6999405 (1)\ttotal: 4m 30s\tremaining: 1m 7s\n",
      "900:\tlearn: 0.6995666\ttest: 0.6992622\tbest: 0.6999405 (1)\ttotal: 5m 5s\tremaining: 33.5s\n",
      "999:\tlearn: 0.6996110\ttest: 0.6993038\tbest: 0.6999405 (1)\ttotal: 5m 37s\tremaining: 0us\n",
      "\n",
      "bestTest = 0.6999405304\n",
      "bestIteration = 1\n",
      "\n",
      "Shrink model to first 2 iterations.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<catboost.core.CatBoostClassifier at 0x17a133cb0>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from catboost import CatBoostClassifier\n",
    "\n",
    "\n",
    "model_cat = CatBoostClassifier(\n",
    "    iterations=1000,\n",
    "    depth=8,\n",
    "    learning_rate=0.05,\n",
    "    loss_function='Logloss',\n",
    "    eval_metric='F1',\n",
    "    class_weights=[1, 1],  # можно задать вручную: [w0, w1]\n",
    "    verbose=100\n",
    ")\n",
    "model_cat.fit(X_train, y_train, eval_set=(X_test, y_test))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "edc3b7f0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0       0.70      0.12      0.21   1658610\n",
      "           1       0.55      0.95      0.70   1883608\n",
      "\n",
      "    accuracy                           0.57   3542218\n",
      "   macro avg       0.63      0.54      0.46   3542218\n",
      "weighted avg       0.62      0.57      0.47   3542218\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Предсказания классов (порог по умолчанию 0.5)\n",
    "y_pred_cat = model_cat.predict(X_test)\n",
    "\n",
    "# Метрики\n",
    "print(classification_report(y_test, y_pred_cat))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0fd39505",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Fitting 3 folds for each of 20 candidates, totalling 60 fits\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_506577d01ecd409bb9e597bc7c7be11c for automatic cleanup: unknown resource type folder\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /loky-27980-fd_fppxo for automatic cleanup: unknown resource type semlock\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /loky-27980-jt2qdgew for automatic cleanup: unknown resource type semlock\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /loky-27980-x6cuqbxt for automatic cleanup: unknown resource type semlock\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /loky-27980-xrdgtvem for automatic cleanup: unknown resource type semlock\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /loky-27980-rxskd5hn for automatic cleanup: unknown resource type semlock\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /loky-27980-0n5jxnl5 for automatic cleanup: unknown resource type semlock\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded for automatic cleanup: unknown resource type folder\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /loky-27980-lcm_zlc1 for automatic cleanup: unknown resource type semlock\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /loky-27980-5u9holkc for automatic cleanup: unknown resource type semlock\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /loky-27980-ii2tku0o for automatic cleanup: unknown resource type semlock\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /loky-27980-3eurfo_t for automatic cleanup: unknown resource type semlock\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /loky-27980-ng24_6fe for automatic cleanup: unknown resource type semlock\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /loky-27980-7w86zv6n for automatic cleanup: unknown resource type semlock\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /loky-27980-8h__vyzz for automatic cleanup: unknown resource type semlock\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /loky-27980-bg_ojkhb for automatic cleanup: unknown resource type semlock\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /loky-27980-94cdt_6u for automatic cleanup: unknown resource type semlock\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /loky-27980-223q4ojb for automatic cleanup: unknown resource type semlock\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /loky-27980-jgath3mz for automatic cleanup: unknown resource type semlock\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /loky-27980-q10557wo for automatic cleanup: unknown resource type semlock\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-77bc42e90e9b45f7b1d010f2b850de87.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-77bc42e90e9b45f7b1d010f2b850de87.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-7219961fc73344868d3bdbe09edff242.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-7219961fc73344868d3bdbe09edff242.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-6338908a96394d119d2f6e95a1fe0cd5.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-6338908a96394d119d2f6e95a1fe0cd5.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-e0ee0972aa9f4ca98d08790d69a71910.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-e0ee0972aa9f4ca98d08790d69a71910.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-8c16d6c6ee834e0094a3b27906759df0.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-8c16d6c6ee834e0094a3b27906759df0.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-dcb3306a4ebb4695a086338938271e10.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-dcb3306a4ebb4695a086338938271e10.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-77bc42e90e9b45f7b1d010f2b850de87.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-7219961fc73344868d3bdbe09edff242.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-6338908a96394d119d2f6e95a1fe0cd5.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-e0ee0972aa9f4ca98d08790d69a71910.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-8c16d6c6ee834e0094a3b27906759df0.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-dcb3306a4ebb4695a086338938271e10.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-77bc42e90e9b45f7b1d010f2b850de87.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-7219961fc73344868d3bdbe09edff242.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-6338908a96394d119d2f6e95a1fe0cd5.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-e0ee0972aa9f4ca98d08790d69a71910.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-8c16d6c6ee834e0094a3b27906759df0.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-dcb3306a4ebb4695a086338938271e10.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-77bc42e90e9b45f7b1d010f2b850de87.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-7219961fc73344868d3bdbe09edff242.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-6338908a96394d119d2f6e95a1fe0cd5.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-e0ee0972aa9f4ca98d08790d69a71910.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-8c16d6c6ee834e0094a3b27906759df0.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-dcb3306a4ebb4695a086338938271e10.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-77bc42e90e9b45f7b1d010f2b850de87.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-7219961fc73344868d3bdbe09edff242.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-6338908a96394d119d2f6e95a1fe0cd5.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-e0ee0972aa9f4ca98d08790d69a71910.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-8c16d6c6ee834e0094a3b27906759df0.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-dcb3306a4ebb4695a086338938271e10.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-77bc42e90e9b45f7b1d010f2b850de87.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-7219961fc73344868d3bdbe09edff242.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-6338908a96394d119d2f6e95a1fe0cd5.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-e0ee0972aa9f4ca98d08790d69a71910.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-8c16d6c6ee834e0094a3b27906759df0.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-dcb3306a4ebb4695a086338938271e10.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-77bc42e90e9b45f7b1d010f2b850de87.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-7219961fc73344868d3bdbe09edff242.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-6338908a96394d119d2f6e95a1fe0cd5.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-e0ee0972aa9f4ca98d08790d69a71910.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-8c16d6c6ee834e0094a3b27906759df0.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-dcb3306a4ebb4695a086338938271e10.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-77bc42e90e9b45f7b1d010f2b850de87.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-7219961fc73344868d3bdbe09edff242.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-6338908a96394d119d2f6e95a1fe0cd5.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-e0ee0972aa9f4ca98d08790d69a71910.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-bd6d1a6785394203ada8fa41480eff79.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-edeb86c519b346bd86b0e6d41fb428d1.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-8c16d6c6ee834e0094a3b27906759df0.pkl for automatic cleanup: unknown resource type file\u001b[0m\n",
      "Traceback (most recent call last):\n",
      "  File \u001b[35m\"/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/multiprocessing/resource_tracker.py\"\u001b[0m, line \u001b[35m295\u001b[0m, in \u001b[35mmain\u001b[0m\n",
      "    raise ValueError(\n",
      "        f'Cannot register {name} for automatic cleanup: '\n",
      "        f'unknown resource type {rtype}')\n",
      "\u001b[1;35mValueError\u001b[0m: \u001b[35mCannot register /var/folders/r0/jlw4c4zs6c540l3_z3d9ch840000gn/T/joblib_memmapping_folder_27980_970ea0c049fc4ffe8ecdac596e9464bb_8210d4ccfd1646278a27d61a3d620ded/27980-6039102064-dcb3306a4ebb4695a086338938271e10.pkl for automatic cleanup: unknown resource type file\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "from catboost import CatBoostClassifier\n",
    "from sklearn.model_selection import RandomizedSearchCV\n",
    "from sklearn.metrics import make_scorer, f1_score\n",
    "\n",
    "# Определяем модель\n",
    "model_cat = CatBoostClassifier(\n",
    "    loss_function='Logloss',\n",
    "    eval_metric='F1',\n",
    "    class_weights=[1, 10],  # предварительные веса, можно оставить None\n",
    "    verbose=0,\n",
    "    random_seed=42\n",
    ")\n",
    "\n",
    "# Гиперпараметры для подбора\n",
    "param_dist = {\n",
    "    'iterations': [300, 500, 800, 1000],\n",
    "    'depth': [4, 6, 8, 10],\n",
    "    'learning_rate': [0.01, 0.03, 0.05, 0.1],\n",
    "    'l2_leaf_reg': [1, 3, 5, 7, 10],\n",
    "    'class_weights': [[1, 5], [1, 10], [1, 20]]\n",
    "}\n",
    "\n",
    "# F1-мера для миноритарного класса (label=1)\n",
    "f1_scorer = make_scorer(f1_score, average='binary', pos_label=1)\n",
    "\n",
    "# RandomizedSearchCV (подбирает случайное подмножество параметров)\n",
    "random_search = RandomizedSearchCV(\n",
    "    estimator=model_cat,\n",
    "    param_distributions=param_dist,\n",
    "    n_iter=20,  # количество случайных комбинаций\n",
    "    scoring=f1_scorer,\n",
    "    cv=3,\n",
    "    verbose=2,\n",
    "    n_jobs=-1\n",
    ")\n",
    "\n",
    "random_search.fit(X_train, y_train)\n",
    "\n",
    "print(\"Лучшие параметры:\", random_search.best_params_)\n",
    "print(\"Лучший F1-score:\", random_search.best_score_)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a80c7fd8",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
